problems/arraysWithLoops.js

```diff
@@ -66,15 +117,45 @@ function findSmallest() {}
  * @returns {number} second smallest value
  */
 
-function findSecondSmallest() {}
-
+function findSecondSmallest(nums) {
+  let smallest = Infinity;
+  let secondSmallest = Infinity;
+  for (let i = 0; i < nums.length; i++) {
+    const num = nums[i];
+    if (num < smallest) {
+      secondSmallest = smallest;
+      smallest = num;
+    } else if (num < secondSmallest) {
+      secondSmallest = num;
+    }
+  }
+  return secondSmallest;
+}
+
+// let smalles = Infinity;
+// let secondSmallest = Infinity;
+
+// for (let i = 0; i < arr.length; i++)
 /**
  * Takes in array and returns the second largest number in the array
  * @param {number[]} nums
  * @returns {number} second largest value
  */
 
-function findSecondLargest() {}
+function findSecondLargest(nums) {
+  let largest = -Infinity;
+  let secondLargest = -Infinity;
+  for (let i = 0; i < nums.length; i++) {
+    const num = nums[i];
+    if (num > largest) {
+      secondLargest = largest;
+      largest = num;
+    } else if (num > secondLargest) {
+      secondLargest = num;
```

Comment:
Good correct for both this and secondSmallest. Keep in mind that your function should always work for all arrays inputted, not just certain cases